src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -790,6 +793,12 @@ public ImmutableArray<T> RemoveAt(int index)
         [Pure]
         public ImmutableArray<T> RemoveRange(int index, int length)
         {
+            // TODO: There is an inconsistency here between what type of exception will be thrown
+            // if we are an IsDefault instance.
+            // If the index is less than 0, the && below will short-circuit, and self.Length will
+            // not be evaluated. Then we will get an ArgumentOutOfRangeException.
+            // Otherwise, we will get a NullReferenceException when trying to evaluate self.Length.
+
             var self = this;
             Requires.Range(index >= 0 && index <= self.Length, nameof(index));
```

Comment:
There is a problem here where if `self` is default and `index` is negative, an ArgumentOutOfRangeException will be thrown. But if `index` is positive or `length` is negative, a NullReferenceException will be thrown. Would it be acceptable to add a `self.ThrowNullRefIfNotInitialized()` before this line?